### PR TITLE
Feat: Dry_run of a workflows

### PR DESCRIPTION
**Pull request by SebastianMorawiec**
# The problem
It would be useful to test the “decorators” of the tasks/workflows.

This could be done by running the workflow and each task is a no-op, but the resources, imports, etc. are all allocated/installed.
(https://zapatacomputing.atlassian.net/browse/ORQSDK-919)

# This PR's solution
Add new option to workflow runs that allows to run workflow without actually running the task code

# Checklist

_Check that this PR satisfies the following items:_

- [x] Tests have been added for new features/changed behavior (if no new features have been added, check the box).
- [x] The [changelog file](CHANGELOG.md) has been updated with a user-readable description of the changes (if the change isn't visible to the user in any way, check the box).
- [x] The PR's title is prefixed with `<feat/fix/chore/imp[rovement]/int[ernal]/docs>[!]:`
- [x] The PR is linked to a JIRA ticket (if there's no suitable ticket, check the box).

### CODE REVIEW

File: src/orquestra/sdk/_base/_driver/_models.py
Comment (alexjuda) [inline src/orquestra/sdk/_base/_driver/_models.py:274]: **issue**: Is it supported on the Platform already? This field isn't in the [CE's API spec](https://github.com/zapatacomputing/workflow-driver/blob/5d030e5a71be088c86590ce63180e256b55b3c8e/openapi/src/schemas/CreateWorkflowRunRequest.yaml). Even if we want to add client code before the Platform is ready, the spec should be updated first.

**Review by alexjuda — CHANGES_REQUESTED** [review]

File: src/orquestra/sdk/_base/_driver/_models.py
Comment (SebastianMorawiec) [reply src/orquestra/sdk/_base/_driver/_models.py:274]: Its not supported - my intention is to wait for workflow-driver to support this feature before I merge this.

File: src/orquestra/sdk/_ray/_build_workflow.py
Comment (alexjuda) [inline src/orquestra/sdk/_ray/_build_workflow.py:170]: **nitpick**: this class (`ArgumentUnwrapper`) is about unwrapping arguments to make it suitable for interfacing with Ray tasks. The `dry_run` is an unrelated feature. Can you extract it to a different component? Functions are composable.

File: tests/runtime/ray/test_integration.py
Comment (alexjuda) [inline tests/runtime/ray/test_integration.py:124]: **nitpick**: please use named arguments when passing literals.

File: tests/sdk/api/test_wf_run.py
Comment (alexjuda) [inline tests/sdk/api/test_wf_run.py:1373]: nitpick: please use named arguments when passing literals.

File: tests/sdk/driver/test_ce_runtime.py
Comment (alexjuda) [inline tests/sdk/driver/test_ce_runtime.py:85]: nitpick: please use named arguments when passing literals.

File: tests/sdk/driver/test_client.py
Comment (alexjuda) [inline tests/sdk/driver/test_client.py:1043]: nitpick: please use named arguments when passing literals.

File: tests/sdk/test_in_process_runtime.py
Comment (alexjuda) [inline tests/sdk/test_in_process_runtime.py:80]: nitpick: please use named arguments when passing literals.

File: tests/runtime/ray/test_build_workflow.py
Comment (alexjuda) [inline tests/runtime/ray/test_build_workflow.py:402]: nitpick: please use named arguments when passing literals.

**Review by alexjuda — CHANGES_REQUESTED** [review]

File: tests/sdk/api/test_wf_run.py
Comment (SebastianMorawiec) [reply tests/sdk/api/test_wf_run.py:1373]: For all the comments about passing literals: I can do changes with function call, cant apply them to mock.asserts - if parameter is passed as arg, it has to be checked as arg, not kwarg

File: src/orquestra/sdk/_ray/_build_workflow.py
Comment (SebastianMorawiec) [reply src/orquestra/sdk/_ray/_build_workflow.py:151]: Ive also checked your example of:
```
@sdk.task
def foo():
    return "abc"
```

- Right now we treat strings as non-subscribtable. So return "Abc" is the same for us as return <any literal like None>

File: src/orquestra/sdk/_ray/_build_workflow.py
Comment (alexjuda) [inline src/orquestra/sdk/_ray/_build_workflow.py:456]: **nitpick**: shouldn't `n_outputs` be replaced with this parameter?

**Review by alexjuda — APPROVED** [review]

**Pull request merged**
